src/node-mailer/node-mailer.service.ts

```diff
@@ -0,0 +1,29 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import * as nodemailer from 'nodemailer';
+
+@Injectable()
+export class NodeMailerService {
+  private transporter: nodemailer.Transporter;
+
+  constructor(@Inject(ConfigService) private configService: ConfigService) {
+    this.transporter = nodemailer.createTransport({
+      service: 'gmail',
+      auth: {
+        user: configService.get<string>('NODEMAILER_USER'),
+        pass: configService.get<string>('NODEMAILER_PASSWORD'),
+      },
+    });
+  }
+
+  async sendEmail(to: string, subject: string, text: string, from = '') {
+    const info = await this.transporter.sendMail({
+      from,
+      to,
+      subject,
+      text,
+    });
+
+    if (!info.messageId) throw new Error('Email not sent');
```

Comment:
where do we handle this ?

Reply:
It's not handled. What do you suggest? Should I try resending the email a predefined number of times before we completely be sure that nothing can be done from our side?

Reply:
I am talking about the response to the user

Reply:
This thrown error will return an error response to the user